### DISCUSSION TITLE
Option to random-sort subfolders into general non-hierarchical random display order

### DISCUSSION THREAD

**Discussion by Chalice** [sylikc/jpegview#154] [Ideas]
Hello!

There is one Option that I am missing from JPEGView, which is pretty much the only reason I am using ArtSage still, at times:

The ability to have the random display order not randomly go through the mainfolder and then get stuck in one of the subfolders, but to essentially have all the subfolder's content's be shuffled into the main random queue.

I.e. an option to, instead of:

(Random Display of Main folder items)
 ->   ( Then random display of Subfolder A..)
 ->    (etc.)

To do:
(Random Display of main- and subfolder items mixed)

Just an idea, but it's pretty much the only thing I am missing from JPEGView! :)

**Comment by sylikc** [comment]
It's an interesting idea... The way JPEGView goes from one folder to the next right now... it would take quite a rework to get this idea feasible.

Currently, when JPEGView arrives at a folder, it does a list of just that folder.  Regardless of whether that folder is using ModDate, FileName, or Random, it creates a list which the it goes through sequentially.  When arriving at the end of that list, then it looks for other folders, and then goes and lists into that.

I'll keep the suggestion in mind, but right now it doesn't list all folders at once, which is sort of necessary to random all folders.
